bin/collator/src/parachain/service.rs

```diff
@@ -42,105 +41,64 @@ use futures::StreamExt;
 use sc_client_api::BlockchainEvents;
 use sc_consensus::{import_queue::BasicQueue, ImportQueue};
 use sc_executor::{HeapAllocStrategy, WasmExecutor, DEFAULT_HEAP_ALLOC_STRATEGY};
-use sc_network::{NetworkBackend, NetworkBlock};
+use sc_network::{config::NetworkBackendType, NetworkBackend, NetworkBlock};
 use sc_network_sync::SyncingService;
 use sc_service::{Configuration, PartialComponents, TFullBackend, TFullClient, TaskManager};
 use sc_telemetry::{Telemetry, TelemetryHandle, TelemetryWorker, TelemetryWorkerHandle};
-use sp_api::{ConstructRuntimeApi, ProvideRuntimeApi};
+use sp_api::{ApiExt, ProvideRuntimeApi};
 use sp_consensus_aura::{
     sr25519::AuthorityId as AuraId, sr25519::AuthorityPair as AuraPair, AuraApi,
 };
 use sp_keystore::KeystorePtr;
-use sp_runtime::{
-    traits::{BlakeTwo256, Block as BlockT},
-    Percent,
-};
+use sp_runtime::{traits::Block as BlockT, Percent};
 use std::{collections::BTreeMap, sync::Arc, time::Duration};
 use substrate_prometheus_endpoint::Registry;
 
 use super::shell_upgrade::*;
 
-#[cfg(feature = "evm-tracing")]
 use crate::{
     evm_tracing_types::{EthApi as EthApiCmd, EvmTracingConfig},
     rpc::tracing,
 };
 
 /// Parachain host functions
-#[cfg(feature = "evm-tracing")]
 pub type HostFunctions = (
     cumulus_client_service::ParachainHostFunctions,
     moonbeam_primitives_ext::moonbeam_ext::HostFunctions,
 );
 
-/// Parachain host functions
-#[cfg(not(feature = "evm-tracing"))]
-pub type HostFunctions = (cumulus_client_service::ParachainHostFunctions,);
-
 /// Parachain executor
 pub type ParachainExecutor = WasmExecutor<HostFunctions>;
 
+type FullClient =
+    TFullClient<Block, crate::parachain::fake_runtime_api::RuntimeApi, ParachainExecutor>;
```

Comment:
Dumb question - with the `crate::parachain::fake_runtime_api::RuntimeApi`, won't this make it a strict requirement that all runtimes provide the _same_ API?

E.g. if we add something to Shibuya that is not supported (yet) by Astar.

Reply:
This is to tell compiler that those traits are implemented on our runtimes. This includes only runtime_apis required by client.